File: nonsoAndrew/day7/subset.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsetsWithDup = (nums) => {
	nums.sort((a, b) => a - b);
	const result = [];
	subsetsUtils(0, []);
	return result;
	function subsetsUtils(index, array) {
		result.push([...array]);
		for (let i = index; i < nums.length; i++) {
			if (i > index && nums[i] == nums[i - 1]) {
				continue;
			}
			array.push(nums[i]);
			subsetsUtils(i + 1, array);
			array.pop();
		}
	}
};

File: solutions/Munachi/subset.py
def subsetsWithDup(nums):
    result = [[]]
    for i in nums:
        for j in range(len(result)):
            x = result[j][:]
            x.append(i)
            x.sort()
            if x not in result:
                result.append(x)
            else:
                continue
    return result